pandas/core/dtypes/base.py

```diff
@@ -236,6 +236,10 @@ def construct_from_string(cls, string: str):
         """
         if not isinstance(string, str):
             raise TypeError(f"Expects a string, got {type(string).__name__}")
+
+        # error: Non-overlapping equality check (left operand type: "str", right
+        #  operand type: "Callable[[ExtensionDtype], str]")  [comparison-overlap]
+        assert isinstance(cls.name, str), (cls, type(cls.name))
```

Comment:
assertion needed for mypy.

base class uses `@property` for abstract class attribute `name`, however, PeriodDType and DatetimeTZDtype do not override this with class attribute for name, but instead create a name using instance properties. These ExtensionDtype subclass cannot use this method and must override it.

Reply:
Is the second expression supposed to be `isinstance` as well?

Reply:
On master BooleanDtype.construct_from_string and StringDtype.construct_from_string both call super. `string != cls.name` is always True since cls.name is a property object. without the changes here would need an ignore.


> Is the second expression

i assume you mean `(cls, type(cls.name))`. that's the error message to identify the classes that call this method when cls.name is not a string